vespalib/src/vespa/vespalib/datastore/simple_hash_map.h

```diff
@@ -0,0 +1,43 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include "atomic_entry_ref.h"
+#include <atomic>
+#include <vespa/vespalib/util/generationholder.h>
+#include <functional>
+
+namespace vespalib::datastore {
+
+class FixedSizeHashMap;
+class EntryComparator;
+
+/*
+ * Hash map over keys in data store.
```

Comment:
Please add more details on the implementation, how it relates to a data store, what is the key and value, and how to use it (e.g. how it relates to writer / readers and generation handling).

Also consider if the class name is descriptive enough.